compliance/aws/untagged_resources/aws_untagged_resources.pt

```diff
@@ -0,0 +1,266 @@
+  code <<-EOS
+  var result=[];
+  for(var i=0;i<ds_get_all_resources.length;i++){
+    var missing_tags="";
+    var all_tags="";
+    var instance=ds_get_all_resources[i];
+    for(var j=0;j<param_tags_to_check.length;j++){
+      var param_lowerCase=param_tags_to_check[j].toLowerCase();
+      var notMatched=true;
+      if(param_lowerCase.indexOf("=")==-1)
+      {
+        for(var k=0;k<instance.tags.length;k++){
+          var current_tag=instance.tags[k]; 
+          if(current_tag["tagKey"].toLowerCase()==param_lowerCase || current_tag["tagValue"].toLowerCase()==param_lowerCase){
+            notMatched=false;
+            break;
+          }
+        }
+      }
+      else {
+        var arr=param_lowerCase.split("=");
+        for(var k=0;k<instance.tags.length;k++){
+          var current_tag=instance.tags[k];
+          if(current_tag["tagKey"].toLowerCase()==arr[0] && current_tag["tagValue"].toLowerCase()==arr[1]){
+            notMatched=false;
+            break;
+          }
+        }
+      }
+      if(notMatched){
+        missing_tags=missing_tags+", "+param_tags_to_check[j];
+      }
+    }
+    for(var j=0;j<instance.tags.length;j++){
+      var current_tag=instance.tags[j];
+      all_tags=all_tags+", "+current_tag["tagKey"]+"="+current_tag["tagValue"];
+    }
+    if(all_tags.length==0){
+      all_tags="< No Value >";
+    }
+    else{
+      all_tags=all_tags.slice(2);
+    }
+    var id="";
+    var arn_last=instance.arn.split(":");
+    if(arn_last[arn_last.length-1].indexOf("/")!=-1){
+      var arr=arn_last[arn_last.length-1].split("/")
+      id=arr[1];
+    }
+    else{
+      id=arn_last[arn_last.length-1];
+    }
+    if(missing_tags.length!=0){
+      result.push({
+        "arn":instance.arn,
+        "id": id,
+        "missing_tags":missing_tags.slice(2),
+        "available_tags":all_tags,
+        "region":instance.region
+      })
+    }
+  }
+  result=_.sortBy(result, 'id');
+  result=_.sortBy(result, 'region');
+  EOS
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+
+escalation "esc_email" do
+  automatic true
+  label "Sends Email"
+  description "Sends Incident Email"
+  email $param_email
+end
+
+escalation "esc_add_tags" do
+  automatic false
+  parameter "param_tags_to_add" do
+    type "list"
+    label "Add Tags Key=Value"
+    description "Add Tags to the selected resource"
+  end
+  run "cwf_add_tags", data, $param_tags_to_add
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_untagged_resources" do
+  validate $ds_filter_resources do
+    summary_template "AWS untagged resources list"
+    escalate $esc_email
+    escalate $esc_add_tags
+    check eq(size(data),0)
+    export do
+      resource_level true
```

Comment:
Is it possible to include the resource type? ie. instance, volume etc?

Reply:
added the resource type column